typing/includecore.ml

```diff
@@ -586,6 +587,10 @@ let report_type_mismatch first second decl env ppf err =
       pr "Their internal representations differ:@ %s %s %s."
          (choose ord first second) decl
          "is extensible"
+  | With_null_representation ord ->
```

Comment:
I'm surprised this didn't change any tests. Should it have? I see that the text is identical to other text, so that helps explain it.... but it also looks like maybe there was a missing case that is now handled here?

Reply:
`or_null` is only made `Variant_with_null` in the final PR that implements the middle-end parts. See the re-export tests in https://github.com/ocaml-flambda/flambda-backend/pull/3267.